plugins/store/guest-js/index.ts

```diff
@@ -8,50 +8,250 @@ import { invoke, Resource } from '@tauri-apps/api/core'
 
 interface ChangePayload<T> {
   path: string
+  resourceId?: number
   key: string
-  value: T | null
+  value: T
+  exists: boolean
 }
 
 /**
  * Options to create a store
  */
 export type StoreOptions = {
   /**
-   * Auto save on modification with debounce duration in milliseconds
+   * Auto save on modification with debounce duration in milliseconds, it's 100ms by default, pass in `false` to disable it
    */
-  autoSave?: boolean
+  autoSave?: boolean | number
+  /**
+   * Name of a serialize function registered in the rust side plugin builder
+   */
+  serializeFnName?: string
+  /**
+   * Name of a deserialize function registered in the rust side plugin builder
+   */
+  deserializeFnName?: string
+}
+
+/**
+ * @param path Path to save the store in `app_data_dir`
+ * @param options Store configuration options
+ *
+ * @throws If a store at that path already exists
+ */
+export async function create(
+  path: string,
+  options?: StoreOptions
+): Promise<Store> {
+  return await Store.create(path, options)
 }
 
 /**
- * @param path: Path to save the store in `app_data_dir`
- * @param options: Store configuration options
+ * Create a new Store or load the existing store with the path.
+ *
+ * If the store at the given path is already loaded,
+ * its instance is returned regardless of the options object.
+ * If the settings to not match an error is returned.
+ *
+ * @param path Path to save the store in `app_data_dir`
+ * @param options Store configuration options
  */
-export async function createStore(path: string, options?: StoreOptions) {
-  const resourceId = await invoke<number>('plugin:store|create_store', {
-    path,
-    ...options
-  })
-  return new Store(resourceId, path)
+export async function newOrExisting(
```

Comment:
I don't quite like this name, I think it should be
```suggestion
export async function createNewOrLoadExisting(
```
or `createNewOrGetExisting`

Reply:
I would probably like `createOrExisting` a bit more like how it was @lucasfernog

Reply:
`createOrExisting` just feels like it's missing a word in there.. omg all these names are awful